keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -232,27 +287,248 @@ demonstrate the interest in a KEP within the wider Kubernetes community.
+
+  *Limitation*: Optional allocation is supported neither by the device
+  plugins nor by current Pod resource declaration.
+
+- *Support Over the Fabric devices*: When deploying a container, I’d
+  like to utilize devices available over the Fabric (network, special
+  links, etc).
+
+  *Limitation*: The device plugin API is designed for node-local resources that
+  get discovered by a plugin running on the node. Projects like
+  [Akri](https://www.cncf.io/projects/akri/) have to work around that by
+  reporting the same network-attached resource on all nodes that it could
+  get attached to and then updating resource availability on all of those
+  nodes when resources get used.
+
+Several other limitations are addressed by
+[CDI](https://github.com/container-orchestrated-devices/container-device-interface/),
+a container runtime extension that this KEP is using to expose resources
+inside a container.
+
 ### Goals
 
 - Enable cluster autoscaling when pods use resource claims, with correct
   decisions and changing the cluster size by more than one node at a time.
 
-- Support node-local resources. Adding or removing nodes has no effect
-  on network-attached resources and therefore CA does not need to (and cannot)
-  simulate them.
+- Support node-local resources
 
-- Allow DRA driver developers to provide a user experience that is similar to
-  the one possible without structured parameters. Ideally, users should not notice
+- Support claim parameters that are specified in a vendor CRD as
+  an alternative to letting users directly specify parameters with
+  the in-tree type. This provides a user experience that is similar to
+  what has been possible since Kubernetes 1.26. Ideally, users should not notice
   at all that a driver is using structured parameters under the hood.
 
 ### Non-Goals
 
-- Scheduling performance is expected to become better compared to using the
-  PodSchedulingContext. However, this is not the reason for this KEP.
-
+* Replace the device plugin API. For resources that fit into its model
+  of a single, linear quantity it is a good solution. Other resources
+  should use dynamic resource allocation. Both are expected to co-exist, with vendors
+  choosing the API that better suits their needs on a case-by-case
+  basis. Because the new API is going to be implemented independently of the
+  existing device plugin support, there's little risk of breaking stable APIs.
+
+* Provide an abstraction layer for resource requests, i.e., something like a
+  “I want some kind of GPU”. Users will need to know about specific
+  resource drivers and which parameters they support. Portability of
+  workloads could be added on top of this proposal by introducing the
+  selection of a resource implementation through labels and
+  standardizing those labels and the associated parameters. The
+  [Resource Class
+  Proposal](https://docs.google.com/document/d/1qKiIVs9AMh2Ua5thhtvWqOqW0MSle_RV3lfriO1Aj6U/edit#heading=h.jzfmfdca34kj)
```

Comment:
Sorry, I don't have control over that document.

Are you subscribed to some of the relevant Google groups? I can't access the document with my private account and can access it with my work account, presumably because this is subscribed to several groups.

Reply:
> Are you subscribed to some of the relevant Google groups?

Apparently not (and I was using work account), but if you do not control the doc, let's forget this.